aea/runner.py

```diff
@@ -0,0 +1,234 @@
+        """
+        Stop particular agent.
+
+        :param agent: AEA instance to stop.
+        :return: None
+        """
+        agent.stop()
+
+
+class AsyncExecutor(BaseAEAExecutor):
+    """Thread based executor to run mul,tiple agents in threads."""
+
+    def _set_executor_pool(self) -> None:
+        """Do nothing, cause we run tasks in asyncio event loop and do not need an executor pool."""
+
+    def _start_agent(self, agent: AEA) -> Awaitable:
+        """
+        Start particular agent.
+
+        :param agent: AEA instance to start.
+        :return: awaitable task to get result or exception
+        """
+        if agent._runtime_mode != "async":
+            raise ValueError(
+                "Agent has to use async runtime mode to run with Async mode runner."
+            )
+        agent._runtime.set_loop(self._loop)
+        return self._loop.create_task(agent._runtime._run_runtime())  # type: ignore # checked above
+
+    def _stop_agent(self, agent: AEA) -> None:
+        """
+        Stop particular agent.
+
+        :param agent: AEA instance to stop.
+        :return: None
+        """
+        agent.stop()
+
+
+class AEARunner:
+    """Run multiple AEA instances."""
+
+    SUPPORTED_MODES: Dict[str, Type[BaseAEAExecutor]] = {
+        "threaded": ThreadExecutor,
+        "async": AsyncExecutor,
+    }
+
+    def __init__(self, agents: Sequence[AEA], mode: str) -> None:
+        """
+        Init AEARunner.
+
+        :param agents: sequence of AEA instances to run.
+        :param mode: executor name to use.
+        """
+        self.agents: Sequence[AEA] = agents
+        if mode not in self.SUPPORTED_MODES:
+            raise ValueError(f"Unsupported mode: {mode}")
+        self._mode: str = mode
+        self._executor: BaseAEAExecutor = self._make_executor(agents, mode)
+        self._thread: Optional[Thread] = None
+
+    def _make_executor(self, agents: Sequence[AEA], mode: str) -> BaseAEAExecutor:
+        """
+        Make an executor instance to run agents with.
+
+        :param agents: AEA instances to  run.
+        :param mode: executor mode to use.
+
+        :return: aea executor instance
+        """
+        executor_cls = self.SUPPORTED_MODES[mode]
+        return executor_cls(agents=agents)
+
+    @property
+    def is_running(self) -> bool:
+        """Return state of the executor."""
+        return self._executor.is_running
+
+    def start(self, threaded: bool = False) -> None:
+        """
+        Run agents.
+
+        :param threaded: run in dedicated thread without blockg current thread.
+
+        :return: None
+        """
+        self._is_running = True
+        if threaded:
+            self._thread = Thread(target=self._executor.start, daemon=True)
+            self._thread.start()
+        else:
+            self._executor.start()
+
+    def stop(self) -> None:
+        """Stop agents."""
+        self._is_running = False
+        self._executor.stop()
```

Comment:
In threaded case we should join the thread?

Reply:
will be added on next commit